lib/SILOptimizer/Mandatory/OwnershipModelEliminator.cpp

```diff
@@ -627,9 +619,40 @@ static bool stripOwnership(SILFunction &func) {
   if (func.isExternalDeclaration())
     return false;
 
+  llvm::DenseMap<PartialApplyInst *, SmallVector<SILInstruction *>>
+      lifetimeEnds;
+
+  // Nonescaping closures are represented ultimately as trivial pointers to
+  // their context, but we use ownership to do borrow checking of their captures
+  // in OSSA. Now that we're eliminating ownership, we need to dealloc_stack the
+  // context. Go over partial_apply [on_stack] and collect its lifetime ends. We
+  // cannot directly create dealloc_stack here because we are still in OSSA.
```

Comment:
I think the purpose of this code would be more clear if you say that the partial_apply's lifetime ends need to be gathered before we begin to leave OSSA but no `dealloc_stack` can be emitted until after we leave OSSA